lib/cchain/provider/provider.go

```diff
@@ -179,9 +188,19 @@ func (p Provider) stream(
 	srcChain := p.chainNamer(chainVer)
 	ctx := log.WithCtx(in, "src_chain", srcChain, "worker", workerName)
 
+	// Cache the previous batch fetch height, to prevent lookback searching each time
+	var cachedOffsetHeight uint64
```

Comment:
nit: rename to `fetchCursor` to align with new comments of interface, since this is a user of the interface and should follow its docs.